sui_core/src/authority.rs

```diff
@@ -190,7 +190,9 @@ impl AuthorityState {
         let transaction_digest = transaction.digest();
 
         // Ensure an idempotent answer.
-        if self._database.transaction_exists(&transaction_digest)? {
+        if self._database.transaction_exists(&transaction_digest)?
+            || self._database.effects_exists(&transaction_digest)?
```

Comment:
do we need one of them or both of them?

Reply:
Good question.
The authority could be in the state where it has seen the tx, signed it, but has not yet processed the cert. And now it sees the same tx again.
This is rare, but in theory could happen?

Reply:
What if if the user/gateway retries and retries before it gets a response (ie due to a timeout)?

Reply:
Maybe I didn't understand your original question. What's the issue if the database will receive only one of them eventually?

Reply:
I don't have a particular issue (or known attack) in mind, but will spend some time next week on exhausting edge cases.